From 2677253d0425624fe169a189f88309cecb592698 Mon Sep 17 00:00:00 2001
From: Sandy Luo <sandy.luo@amlogic.com>
Date: Fri, 13 Mar 2020 19:22:52 +0800
Subject: [PATCH] ddr: set ro.first.boot.ddr.window property based on bootargs
 [1/2]

PD#SWPL-22113

PD#SWPL-22113

Problem:
In some cases, the software needs to do DDR window testing

Solution:
Provides the ro.first.boot.ddr.window attribute to STR for standby use

Verify:
marconi

Change-Id: I68d64a23e1ff21c1d2c989094bb1dd2a1d5c5bcb
Signed-off-by: Sandy Luo <sandy.luo@amlogic.com>
---
 init/init.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/init/init.cpp b/init/init.cpp
index 9c6dfaf..9c4a142 100644
--- a/init/init.cpp
+++ b/init/init.cpp
@@ -358,6 +358,9 @@ static void import_kernel_nv(const std::string& key, const std::string& value, b
         property_set("ro.boot." + key.substr(12), value);
     } else if (key == "mem_size") {
         property_set("ro.mem_size", value);
+    } else if (key == "need_ddr_window_test") {
+        LOG(INFO) << "import_kernel_nv key=" << key << ",value=" << value;
+        property_set("ro.first.boot.ddr.window", value);
     }
 }
 
-- 
2.7.4

